app/main.py

```diff
@@ -1 +1,33 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    def death(self) -> None:
+        if self.health <= 0:
+            Animal.alive.remove(self)
+
+
+class Carnivore(Animal):
+    def bite(self, other: Animal) -> None:
+        if not other.hidden and not isinstance(other, Carnivore):
```

Comment:
```suggestion
        if not other.hidden and isinstance(other, Herbivore):
```
Due to task we can only bite Herbivore